src/ripple/app/misc/SHAMapStoreImp.cpp

```diff
@@ -664,11 +664,11 @@ SHAMapStoreImp::clearPrior (LedgerIndex lastRotated)
                 << deleteBatch << " rows.";
             long long totalRowsAffected = 0;
             long long rowsAffected;
-            soci::statement st = [&]
+            boost::optional<soci::statement> st;
             {
                 auto db = ledgerDb_->checkoutDb();
-                return (db->prepare << deleteQuery);
-            }();
+                st.emplace(db->prepare << deleteQuery);
```

Comment:
`boost::optional` is usually used for nullable objects. We're using it for a different purpose here - to move the point of construction. It's not that `boost::optional` is exotic, but the way we're using it is unusual. 

The lambda solution fails when we don't get RVO and the implementation of `statement` changes so the copy constructor needs to happen under a lock. It's pretty safe.

I still prefer the lambda, but it's a minor issue (and I can certainly follow the code as written). I'll give a thumbs up as is.